lib/content-services/src/lib/common/services/saved-searches.service.ts

```diff
@@ -0,0 +1,138 @@
+ * limitations under the License.
+ */
+
+import { NodesApi, NodeEntry, SearchApi, SEARCH_LANGUAGE, ResultSetPaging } from '@alfresco/js-api';
+import { Injectable } from '@angular/core';
+import { Observable, of, from, ReplaySubject, throwError } from 'rxjs';
+import { catchError, concatMap, first, map, switchMap, take, tap } from 'rxjs/operators';
+import { AlfrescoApiService } from '../../services/alfresco-api.service';
+import { SavedSearch } from '../interfaces/saved-search.interface';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SavedSearchesService {
+    private _searchApi: SearchApi;
+    get searchApi(): SearchApi {
+        this._searchApi = this._searchApi ?? new SearchApi(this.apiService.getInstance());
+        return this._searchApi;
+    }
+
+    private _nodesApi: NodesApi;
+    get nodesApi(): NodesApi {
+        this._nodesApi = this._nodesApi ?? new NodesApi(this.apiService.getInstance());
+        return this._nodesApi;
+    }
+
+    private savedSearchFileNodeId: string;
+    private createFileAttempt = false;
+
+    readonly SAVED_SEARCHES_NODE_ID = 'saved-searches-node-id';
+
+    savedSearches$ = new ReplaySubject<SavedSearch[]>(1);
+
+    constructor(private apiService: AlfrescoApiService) {
+        this.getSavedSearches()
+            .pipe(take(1))
+            .subscribe((searches) => this.savedSearches$.next(searches));
+    }
+
+    getSavedSearches(): Observable<SavedSearch[]> {
+        return this.getSavedSearchesNodeId().pipe(
+            concatMap((nodeId) => {
+                this.savedSearchFileNodeId = nodeId;
+                return from(
+                    this.nodesApi.getNodeContent(nodeId).then((content) => {
+                        return this.mapFileContentToSavedSearches(content);
+                    })
+                ).pipe(
+                    catchError((error) => {
+                        if (!this.createFileAttempt) {
+                            this.createFileAttempt = true;
+                            localStorage.removeItem(this.SAVED_SEARCHES_NODE_ID);
+                            return this.getSavedSearches();
+                        }
+                        return throwError(error);
+                    })
+                );
+            })
+        );
+    }
+
+    saveSearch(newSaveSearch: Pick<SavedSearch, 'name' | 'description' | 'encodedUrl'>): Observable<NodeEntry> {
+        return this.getSavedSearches().pipe(
+            take(1),
+            switchMap((savedSearches: Array<SavedSearch>) => {
+                const updatedSavedSearches = [...savedSearches, { ...newSaveSearch, order: savedSearches.length }];
+                return from(this.nodesApi.updateNodeContent(this.savedSearchFileNodeId, JSON.stringify(updatedSavedSearches))).pipe(
+                    tap(() => this.savedSearches$.next(updatedSavedSearches))
+                );
+            })
+        );
+    }
+
+    getSavedSearchesNodeId(): Observable<string> {
```

Comment:
Missing unit tests for this function

Reply:
actually I changed it for a private method as it's only for internal use in this class, so I think getSavedSearches unit tests are covering it's correct usage and there's no need for additional unit tests. Please let me know if you think differently.